src/topology/subset_properties.lean

```diff
@@ -478,57 +494,103 @@ let ⟨m, hm, hsm, hmm⟩ := zorn.zorn_subset₀ { t : set α | is_irreducible t
 
 /-- A maximal irreducible set that contains a given point. -/
 def irreducible_component (x : α) : set α :=
-classical.some (exists_irreducible {x} is_irreducible_singleton)
+classical.some (exists_preirreducible {x} is_irreducible_singleton.is_preirreducible)
 
-theorem is_irreducible_irreducible_component {x : α} : is_irreducible (irreducible_component x) :=
-(classical.some_spec (exists_irreducible {x} is_irreducible_singleton)).1
+lemma irreducible_component_property (x : α) :
+  is_preirreducible (irreducible_component x) ∧ {x} ⊆ (irreducible_component x) ∧
+  ∀ u, is_preirreducible u → (irreducible_component x) ⊆ u → u = (irreducible_component x) :=
+classical.some_spec (exists_preirreducible {x} is_irreducible_singleton.is_preirreducible)
 
 theorem mem_irreducible_component {x : α} : x ∈ irreducible_component x :=
-singleton_subset_iff.1
-  (classical.some_spec (exists_irreducible {x} is_irreducible_singleton)).2.1
+singleton_subset_iff.1 (irreducible_component_property x).2.1
+
+theorem is_irreducible_irreducible_component {x : α} : is_irreducible (irreducible_component x) :=
+⟨⟨x, mem_irreducible_component⟩, (irreducible_component_property x).1⟩
 
 theorem eq_irreducible_component {x : α} :
-  ∀ {s : set α}, is_irreducible s → irreducible_component x ⊆ s → s = irreducible_component x :=
-(classical.some_spec (exists_irreducible {x} is_irreducible_singleton)).2.2
+  ∀ {s : set α}, is_preirreducible s → irreducible_component x ⊆ s → s = irreducible_component x :=
+(irreducible_component_property x).2.2
 
 theorem is_closed_irreducible_component {x : α} :
   is_closed (irreducible_component x) :=
 closure_eq_iff_is_closed.1 $ eq_irreducible_component
-  (is_irreducible_closure is_irreducible_irreducible_component)
+  (is_irreducible_closure is_irreducible_irreducible_component).is_preirreducible
   subset_closure
 
-/-- An irreducible space is one where there is no non-trivial pair of disjoint opens. -/
-class irreducible_space (α : Type u) [topological_space α] : Prop :=
-(is_irreducible_univ : is_irreducible (univ : set α))
+/-- An preirreducible space is one where there is no non-trivial pair of disjoint opens. -/
+class preirreducible_space (α : Type u) [topological_space α] : Prop :=
+(is_preirreducible_univ : is_preirreducible (univ : set α))
 
-theorem irreducible_exists_mem_inter [irreducible_space α] {s t : set α} :
-  is_open s → is_open t → (∃ x, x ∈ s) → (∃ x, x ∈ t) → ∃ x, x ∈ s ∩ t :=
+/-- An irreducible space is one that is nonempty
+and where there is no non-trivial pair of disjoint opens. -/
+class irreducible_space (α : Type u) [topological_space α]
```

Comment:
Why don't you use a typeclass assumption `[nonempty α]` instead?

Reply:
This has been discussed elsewhere.